config/swap/swap.go

```diff
@@ -0,0 +1,160 @@
+package swap
+
+import (
+	"fmt"
+	"github.com/godbus/dbus/v5"
+	"github.com/godbus/dbus/v5/introspect"
+	"github.com/godbus/dbus/v5/prop"
+	"github.com/home-assistant/os-agent/utils/lineinfile"
+	logging "github.com/home-assistant/os-agent/utils/log"
+	"regexp"
+	"strconv"
+)
+
+const (
+	objectPath        = "/io/hass/os/Config/Swap"
+	ifaceName         = "io.hass.os.Config.Swap"
+	swapPath          = "/etc/default/haos-swapfile"
+	swappinessPath    = "/etc/sysctl.d/15-swappiness.conf"
+	defaultSwappiness = 1
```

Comment:
That seems to be the default in the config file, but not the default default of the kernel :thinking: 

Should we maybe just read the current value using sysctl instead in case the file doesn't exist? :thinking:

Reply:
I was aware of this. The kernel default would never apply as HAOS sets it in read-only part of the FS but I haven't taken Supervised in account. I would have preferred to return it empty too as the "default" but as it's int, it's not nullable. The other drawback is that once you get this default, first call to set the value will persist it in the `/etc/sysctl.d` override.

If you don't have better idea, we could also read `/proc/sys/vm/swappiness` (as it has less overhead than calling `systctl`).

Reply:
> If you don't have better idea, we could also read `/proc/sys/vm/swappiness` (as it has less overhead than calling `systctl`).

Yeah I think that is quite a good and portable solution. If "our" config file isn't there, obviously that is the how the system defaults to through whatever other means it has been set. :+1: